### PR TITLE
Fixes from SCA issues

### PR DESCRIPTION
**Pull request by dalgaaf**

### CODE REVIEW

File: src/crush/CrushWrapper.cc [hunk 1]
```diff
@@ -1850,6 +1850,7 @@ int CrushWrapper::add_bucket(
       crush_choose_arg& carg = cmap.args[pos];
       carg.weight_set = static_cast<crush_weight_set*>(calloc(sizeof(crush_weight_set),
 						  size));
+      assert(carg.weight_set);
```
Comment (tchaikov) [inline src/crush/CrushWrapper.cc]: nit, this change does not belong to this commit.

**Comment by tchaikov** [comment]
@dalgaaf ping?

**Comment by dalgaaf** [comment]
@tchaikov Sorry, have been busy with other projects. I will take a look at it.

**Comment by liewegas** [comment]
@tchaikov ping

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-06-27_11:05:15-rados-wip-kefu-testing-2018-06-27-1254-distro-basic-smithi/
